src/Classes/ServiceAPI/MyRadio_Officer.php

```diff
@@ -265,9 +445,247 @@ public function getCurrentHolders()
     }
 
     /**
+     * Returns all the officer's active permission flags
+     * @return Array
+     */
+    public function getPermissions()
+    {
+        return $this->permissions;
+    }
+
+    /**
+     * Adds a permission flag to the officer
+     * @param $permissionid the permission to add
+     */
+    public function addPermission($permissionid)
+    {
+        self::$db->query(
+            'INSERT INTO public.auth_officer
+            (officerid, lookupid)
+            VALUES ($1, $2)',
+            [$this->getID(), $permissionid]
+        );
+        $this->updateCacheObject();
```

Comment:
my thinking currently is just to update the whole permissions array - as finding a value in an array in an array seems more complex.